### DISCUSSION TITLE
Flash execute javascript function

### DISCUSSION THREAD

**Discussion by juicelink77** [ruffle-rs/ruffle#9132] [Q&A]
Hi,

Thanks for Ruffle is the flash revival, I'm so happy !
I want send a message with actionscript to my html page for switch the full screen.

here my AS code 
```
if (fscommand("EnterFullScreen",fullscreen == true ? "1" : "0")){

}
```
here my js code :
```
function EnterFullScreen (value){
    console.log("EnterFullScreen = " + value);
    window.RufflePlayer.setFullscreen(value == "1" ? true : false);
  }  
```

How I can use the ruffle onFSCommand to call the js function ?
thank you very much.
Julien

**Comment by Lord-McSweeney** [comment]
I think you might want to look into `ExternalInterface` rather than `fscommand`. `ExternalInterface` lets you interact with the Javascript of a page, while `fscommand` can only run a few built-in commands for more common functions, such as `trapallkeys`.

**Reply by juicelink77** [reply]
Hi !
I will test that thank you !

**Reply by juicelink77** [reply]
Don't works :(

Flash : 
`var feedBack = ExternalInterface.call("EnterFullScreen");`

JS :
```
function EnterFullScreen (){
      return "EnterFullScreen OK";
}  
```
config :
```
window.RufflePlayer.config = {
    // Options affecting the whole page
    "allowScriptAccess": true,
    ...
```

**Reply by n0samu** [reply]
`allowScriptAccess` cannot be set from `window.RufflePlayer.config`. If you are using a Flash embed and polyfilling it with Ruffle, you need to add it as an attribute of your `embed` and/or `object` tag - [see this help article](https://helpx.adobe.com/flash/kb/control-access-scripts-host-web.html) for details. If you are using Ruffle's JavaScript API, you need to set it in the `player.load` call - [see our wiki](https://github.com/ruffle-rs/ruffle/wiki/Using-Ruffle#advanced-usage).

**Reply by juicelink77** [reply]
Yeeees I found that few hours ago, it's work now !
thank you for your reply !
